test/src/MadgraphTest.cc

```diff
@@ -0,0 +1,159 @@
+/*
+ * MadgraphTest.cc
+ *
+ *  Created on: 11.12.2020
+ *      Author: shageboeck
+ */
+
+#include "MadgraphTest.h"
+
+#include <cmath>
+#include <fstream>
+#include <sstream>
+
+std::map<unsigned int, ReferenceData> readReferenceData(const std::string& refFileName)
+{
+  std::ifstream referenceFile(refFileName.c_str());
+  EXPECT_TRUE(referenceFile.is_open()) << refFileName;
+  std::map<unsigned int, ReferenceData> referenceData;
+  unsigned int evtNo;
+  unsigned int batchNo;
+
+  for (std::string line; std::getline(referenceFile, line); )
+  {
+    std::stringstream lineStr(line);
+    if (line.empty() || line[0] == '#')
+    {
+      continue;
+    }
+    else if (line.find("Event") != std::string::npos)
+    {
+      std::string dummy;
+      lineStr >> dummy >> evtNo >> dummy >> batchNo;
+    }
+    else if (line.find("ME") != std::string::npos)
+    {
+      if (evtNo <= referenceData[batchNo].MEs.size())
+        referenceData[batchNo].MEs.resize(evtNo + 1);
+
+      std::string dummy;
+      lineStr >> dummy >> referenceData[batchNo].MEs[evtNo];
+    }
+    else
+    {
+      unsigned int particleIndex;
+      lineStr >> particleIndex;
+
+      if (evtNo <= referenceData[batchNo].momenta.size())
+        referenceData[batchNo].momenta.resize(evtNo + 1);
+      if (particleIndex <= referenceData[batchNo].momenta[evtNo].size())
+        referenceData[batchNo].momenta[evtNo].resize(particleIndex + 1);
+
+      auto& fourVec = referenceData[batchNo].momenta[evtNo][particleIndex];
+      for (unsigned int i=0; i < fourVec.size(); ++i) {
+        EXPECT_TRUE(lineStr.good());
+        lineStr >> fourVec[i];
+      }
+      EXPECT_TRUE(lineStr.eof());
+    }
+  }
+  return referenceData;
+}
+
+
+TEST_P(MadgraphTestDouble, eemumu)
+{
+  // Set to dump events:
+  constexpr bool dumpEvents = false;
+  constexpr fptype toleranceMomenta = std::is_same<fptype, double>::value ? 5.E-12 : 1.E-5;
```

Comment:
Same here, could the precision be taken out of the source and made available in a config file? (issue for now is fine)